src/rust/engine/src/scheduler.rs

```diff
@@ -82,7 +82,7 @@ struct InnerSession {
   // The unique id for this Session: used for metrics gathering purposes.
   build_id: String,
   // Per-Session values that have been set for this session.
-  session_values: Mutex<Value>,
```

Comment:
It's odd for this to be anything other than a `Value`, since the API does not support setting an exception here. The `SessionValues` node can do the wrapping (or the `Select` node).